### PR TITLE
Servo pull3

### PR DESCRIPTION
**Pull request by fpvaspassion**
This is a implementation of Servo control mission item. To control PWM outputs used actuators topic, accordingly to developpers strategy. To use it is necessary to upload a particular mixer, from existing mixers mixer sk450_deadcat.aux.mix can be used with correction of limits nad scales.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Awesome! This is looking good from a first glance. Will do code review on this weekend and cross-test it.

**Comment by LorenzMeier** [comment]
Did you test this - do you have a video or something?

**Comment by fpvaspassion** [comment]
Lorenz, i tested it with servo attached, and now i preparing a short manual for this, and i can repeat a test with recording.

**Comment by LorenzMeier** [comment]
Manual in the wiki and video would be awesome! We all love when things move in videos 8).

**Comment by fpvaspassion** [comment]
Lorenz, also question - can i add a mixer for this case into list of predefined mixers?

**Comment by LorenzMeier** [comment]
sure

File: src/modules/navigator/mission_feasibility_checker.cpp [hunk 1]
```diff
@@ -302,9 +303,25 @@ MissionFeasibilityChecker::check_dist_1wp(dm_item_t dm_current, size_t nMissionI
...
+
+					//check actuator number
+			        if (mission_item.actuator_num<1 || mission_item.actuator_num>6) {
+			        	mavlink_log_critical(_mavlink_fd, "Actuator number %d is out of bounds 1..6", (int)mission_item.actuator_num);
+			        	warning_issued = true;
+			        	return false;
+			        }
+					//check actuator value
+			        if (mission_item.actuator_value<900 || mission_item.actuator_value>2000) {
+			        	mavlink_log_critical(_mavlink_fd, "Actuator number %d is out of bounds 900..2000", (int)mission_item.actuator_value);
```
Comment (mhkabir) [inline src/modules/navigator/mission_feasibility_checker.cpp]: *value

**Comment by fpvaspassion** [comment]
Hi,
  Juste tested some times , i attach a link to archive with following:
   etc folder with extras and mixer file
   mission, 
   log file 
   txt with messages from QGC.
Two things are strange for me 
- sdlog do not use GPS time at all,
- wpm error message - it is from function read_mission_item in mission.cpp in navigator, unfortunately i do not understand the purpose of FOR cycle where it appear.
Do you had an opportunity to test this PR? 

https://cloud.mail.ru/public/Hcg9/kWVgJUDcG
  Hi,
  Juste tested some times , i attach a link to archive with following:
   etc folder with extras and mixer file
   mission, 
   log file 
   txt with messages from QGC.
Two things are strange for me 
- sdlog do not use GPS time at all,
- wpm error message - it is from function read_mission_item in mission.cpp in navigator, unfortunately i do not understand the purpose of FOR cycle where it appear.
Do you had an opportunity to test this PR? 

https://cloud.mail.ru/public/Hcg9/kWVgJUDcG
Also link to video http://www.youtube.com/watch?v=Vj1No-rTmmc

**Comment by LorenzMeier** [comment]
Thanks! I've seen that error message as well and I believe it is not an actual error, but I need to spend some time on it.

**Comment by LorenzMeier** [comment]
I will rebase this and apply to master. Please update from master after I do so you have the same version for your further work.

**Comment by LorenzMeier** [comment]
Applied to master. Thanks!

**Pull request closed without merging**